pkg/kubernetes/run.go

```diff
@@ -139,8 +140,11 @@ func (k *kubernetesDriver) runContainer(
 
 	// service account
 	serviceAccount := ""
-	if k.options.ServiceAccount != "" {
+	if k.options.ServiceAccount != "" || k.options.KubernetesPullSecretsEnabled == "true" {
```

Comment:
I don't think this is correct, as it will try to create the default service account here

Reply:
Hmmm, we need a service account for pull secrets (we need to use that SA for pods using this container registry). 

What would be your approach to ensure we have **some** service account here?